### PR TITLE
Moddataimp 41

### PR DESCRIPTION
**Pull request by OleksiiKuzminov**
Changes:

add ret call to mod-source-record-manager
replace jobExecution ids from a response
add mocks for API tests
-bugfixing
add blocking update mechanism

### CODE REVIEW

File: src/main/java/org/folio/dao/UploadDefinitionDaoImpl.java [hunk 1]
```diff
@@ -25,13 +26,116 @@
...
+  @FunctionalInterface
+  public interface UploadDefinitionMutator {
+    /**
+     * @param definition - Loaded from DB UploadDefinition
+     * @return - changed Upload Definition ready for save into database
+     */
+    Future<UploadDefinition> mutate(UploadDefinition definition);
+  }
+
+  public Future<UploadDefinition> updateBlocking(String uploadDefinitionId, UploadDefinitionMutator mutator) {
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/dao/UploadDefinitionDaoImpl.java]: plz, take a look on this update method

File: src/main/java/org/folio/service/file/FileServiceImpl.java [hunk 2]
```diff
@@ -32,30 +33,32 @@ public FileServiceImpl(Vertx vertx, String tenantId, UploadDefinitionService upl
...
-                    .withFileDefinitions(replaceFile(def.getFileDefinitions(), savedFile))
-                    .withStatus(UploadDefinition.Status.IN_PROGRESS))
-                ).orElseThrow(NotFoundException::new))))
-        )
-        .orElseThrow(NotFoundException::new))
-      .compose(reply -> reply)
-      .compose(fileReply -> fileReply)
-      .compose(defReply -> defReply)
-      .compose(result -> result);
+  public Future<UploadDefinition> uploadFile(String fileId, String uploadDefinitionId, InputStream data, OkapiConnectionParams params) {
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/service/file/FileServiceImpl.java]: and how it works now

File: src/main/java/org/folio/dao/UploadDefinitionDaoImpl.java (hunk 1 above)
Comment (Igor-Gorchakov) [reply src/main/java/org/folio/dao/UploadDefinitionDaoImpl.java]: Maybe updateExclusive ? , coz this is exclusive mode, which means the second select cannot proceed until the first one has completed or rolled back.

File: src/main/java/org/folio/service/upload/UploadDefinitionService.java [hunk 3]
```diff
@@ -47,6 +49,15 @@
    */
   Future<UploadDefinition> updateUploadDefinition(UploadDefinition uploadDefinition);
 
+  /**
+   * Updates {@link UploadDefinition} in database with row blocking
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/upload/UploadDefinitionService.java]: Please add java doc about exclusive mode, which means the second select cannot be executed until the first one has completed or rolled back.

**Pull request merged**